### ISSUE TITLE
Check code of ComputeSourceDir and ComputeTestDir

### ISSUE THREAD

**Issue by surli** [Spirals-Team/repairnator#518]
It seems that the code is not quite good neither well tested in both steps!

**Comment by fermadeiral** [comment]
Addition to this issue: to remove duplicate code between the two steps.

**Comment by fermadeiral** [comment]
Project started for testing with multi-module project: https://github.com/fermadeiral/test-repairnator-bears

**Comment by fermadeiral** [comment]
We have several opened issues related to ComputeSourceDir and ComputeTestDir steps, and I'm starting to be lost with the current needs we have, because those steps have been modified, new tests have also been added. I feel we need to "refactor the issues" related to these steps, maybe to centralize them in one issue with a list of specific issues or something like this, I'm not sure how yet, but I believe something needs to change here...WDYT?

**Comment by surli** [comment]
Actually I only see two more issues with this one: 
  - this one: is really abstract and says that we should be careful and check the code of those steps. For me we should keep it open for the moment as it still need to be more tested and/or refactored.
  - #652 asks for a very clear unit test to add
  - #662 this one is not clear for me, so maybe you can add some explanation or close it if it's already done

**Comment by fermadeiral** [comment]
> Actually I only see two more issues with this one:

Yes, I see the same thing as you; for me, 3 related issues characterizes "several".

> this one: is really abstract and says that we should be careful and check the code of those steps. For me we should keep it open for the moment as it still need to be more tested and/or refactored.

Good for me to keep it opened. In fact we might discussed on it also because of the duplicate code, and I think the field `allModules` in `ComputeSourceDir` is also not clear for the moment, because everytime I need to look at this step, I have to understand again and again what this field is for (I'm thinking on putting some doc for it, maybe also renaming this field, I don't know yet...)

> #652 asks for a very clear unit test to add

Ok.

> #662 this one is not clear for me, so maybe you can add some explanation or close it if it's already done

Oh, ok. I didn't put explanation on it because I thought you would understand it since a (closed) PR already existed for it and you also reviewed it, but I will add that now. This is not done yet.

**Comment by surli** [comment]
>  I think the field allModules in ComputeSourceDir is also not clear for the moment

I completely agree

> (I'm thinking on putting some doc for it, maybe also renaming this field, I don't know yet...)

both changes seem like a good idea ;)

> I thought you would understand it since a (closed) PR already existed

yeah actually I might have known about that but I'm really quick to forget :p
